packages/proxy/src/util/responseHelpers.ts

```diff
@@ -0,0 +1,54 @@
+import type { ServerResponse } from 'node:http';
+import { DiscordAPIError, HTTPError, parseResponse, RateLimitError } from '@discordjs/rest';
+import type { Dispatcher } from 'undici';
+
+/**
+ * Populates a server response with the data from a Discord 2xx REST response
+ * @param res The server response to populate
+ * @param data The data to populate the response with
+ */
+export async function populateOkResponse(res: ServerResponse, data: Dispatcher.ResponseData): Promise<void> {
+	res.statusCode = data.statusCode;
+
+	for (const header of Object.keys(data.headers)) {
+		// Strip ratelimit headers
+		if (header.startsWith('x-ratelimit')) {
+			continue;
+		}
+
+		res.setHeader(header, data.headers[header]!);
+	}
+
+	const parsed = await parseResponse(data);
+	res.write(data.headers['content-type']?.startsWith('application/json') ? JSON.stringify(parsed) : parsed);
```

Comment:
I suppose... there's no way to pass raw data back? It would be even better to be able to pass the readable body, similar to this:

Assuming `data` is a `Readable`, which at least the body in HTTP requests are:
```javascript
for await (const chunk of data) res.write(chunk);
```

Reply:
Couldn't you just pipe the response directly?

Reply:
I believe you mean `data.body`?

Reply:
Oh! My bad. GitHub didn't show those comments re piping until I refreshed. Let me have a look at that.